controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

# joint_trajectory_controller:
#   ros__parameters:
#     type: joint_trajectory_controller/JointTrajectoryController
#     joints:
#       - joint1
#       - joint2
#       - joint3
#     command_interfaces:
#       - effort
#     state_interfaces:
#       - position
#       - velocity
#       # - acceleration 

lqr_arm_controller:
  ros__parameters:
    type: lqr_arm_controller/LQRArmController
    joints:
      - joint1
      - joint2
      - joint3
    command_interfaces:
      - effort
    state_interfaces:
      - position
      - velocity
    K: [6.0, 0.0, 0.0, 1.8, 0.0, 0.0,   # joint1: 60*angle + 18*velocity
        0.0, 5.0, 0.0, 0.0, 1.6, 0.0,   # joint2: 50*angle + 16*velocity
        0.0, 0.0, 3.5, 0.0, 0.0, 1.2]   # joint3: 35*angle + 12*velocity
    # K : [0.00128932, 0.000396778, 0.000028487, 0.72431, 0.469444, 0.256659, 
    #      0.000526238, 0.000678042, 0.000265799, 0.477284, 0.368597, 0.21259, 
    #      0.0000306206, 0.000526798, 0.000656808, 0.264048, 0.231906, 0.196494]




#     KposAccel: [4.43654, 2.11726, 0.637776, 26.5045, 17.2688, 8.28275,   # joint1: 60*angle + 18*velocity
#         0.943758, 0.757951, 0.227709, 5.71546, 4.17311, 2.49787,   # joint2: 50*angle + 16*velocity
#         -1.65591, -0.77776, -0.0567327, -9.99926, -5.91985, -2.06833]   # joint3: 35*angle + 12*velocity

base_trajectory_controller:
  ros__parameters:
    type: joint_trajectory_controller/JointTrajectoryController
    joints:
      - world_to_base
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity


    # Working Version from demo:
# controller_manager:
#   ros__parameters:
#     update_rate: 1000  # Hz

#     joint_state_broadcaster:
#       type: joint_state_broadcaster/JointStateBroadcaster

# joint_trajectory_controller:
#   ros__parameters:
#     type: joint_trajectory_controller/JointTrajectoryController
#     joints:
#       - slider_to_cart
#     command_interfaces:
#       - effort
#     state_interfaces:
#       - position
#       - velocity
#     gains:
#       slider_to_cart:
#         p: 100.0
#         i: 0.0
#         d: 10.0
#         i_clamp: 0.0
#         antiwindup: false
